packages/tic-tac-toe/contracts/TicTacToe.sol

```diff
@@ -0,0 +1,617 @@
+        TTTData memory toGameData
+    )
+        private
+        pure
+        noDisjointMoves(toGameData)
+        stakeUnchanged(fromGameData, toGameData)
+        allocationsNotLessThanStake(fromAllocation, toAllocation, fromGameData, toGameData)
+    {
+        require(toGameData.Os == fromGameData.Os, 'No Os added to board');
+        require(madeStrictlyOneMark(toGameData.Xs, fromGameData.Xs), 'One X placed');
+
+        // Current X Player should get all the stake. This is to decrease griefing. We assume that X Player is Player A
+        require(
+            toAllocation[0].amount == fromAllocation[0].amount.add(toGameData.stake*2),
+            'Allocation for player A should be incremented by 1x stake'
+        );
+        require(
+            toAllocation[1].amount == fromAllocation[1].amount.sub(toGameData.stake*2),
+            'Allocation for player B should be decremented by 1x stake.'
+        );
+        
+        // Old TTTMagmo code
+        // if (State.indexOfMover(_new) == 0) { // mover is A
+        //     require(_new.aResolution() == _old.aResolution() + 2 * _new.stake()); // note extra factor of 2 to swing fully to other player
+        //     require(_new.bResolution() == _old.bResolution() - 2 * _new.stake());
+        // } else if (State.indexOfMover(_new) == 1) { // mover is B
+        //     require(_new.aResolution() == _old.aResolution() - 2 * _new.stake());
+        //     require(_new.bResolution() == _old.bResolution() + 2 * _new.stake());
+        // } // mover gets to claim stakes: note factor of 2 to swing fully to other player
+
+    }
+
+    function requireValidXPLAYINGtoVICTORY(
+        Outcome.AllocationItem[] memory fromAllocation,
+        Outcome.AllocationItem[] memory toAllocation,
+        TTTData memory fromGameData,
+        TTTData memory toGameData
+    )
+        private
+        pure
+        noDisjointMoves(toGameData)
+        stakeUnchanged(fromGameData, toGameData)
+    {
+        require(toGameData.Xs == fromGameData.Xs, 'No Xs added to board');
+        require(madeStrictlyOneMark(toGameData.Os, fromGameData.Os), 'One O placed');
+        require(hasWon(toGameData.Os), 'O has won');
+
+        uint256 currentOsPlayer = 1; // Need to calculate this
+
+        uint256 playerAWinnings; // playerOneWinnings
+        uint256 playerBWinnings; // playerTwoWinnings
+        // calculate winnings
+        (playerAWinnings, playerBWinnings) = winnings(currentOsPlayer, toGameData.stake);
+
+        // TODO This logic will only work if PlayerA is Xs and PlayerB is Os
+        require(
+            toAllocation[1].amount == fromAllocation[1].amount.add(playerBWinnings),
+            "Player B's allocation should be updated with the winnings."
+        );
+        require(
+            toAllocation[0].amount ==
+                fromAllocation[0].amount.sub(fromGameData.stake.mul(2)).add(playerAWinnings),
+            "Player A's allocation should be updated with the winnings."
+        );
```

Comment:
You might consider a refactor along the lines of https://github.com/statechannels/monorepo/pull/845/commits/f582c8a2b54f1bb3719de4162dc6349535b109bb to increase clarity.